jira_automation/src/main/java/io/dockstore/jira/Utils.java

```diff
@@ -1,23 +1,111 @@
+
+
     private Utils() { }
 
+    public static JiraIssue getJiraIssue(String issueId)
+        throws URISyntaxException, IOException, InterruptedException {
+        final URI uri = getJiraIssueRestUri(issueId);
+        final HttpRequest httpRequest = authorizedRequestBuilder()
+            .GET()
+            .uri(uri)
+            .build();
+        final HttpResponse<String> response = HTTP_CLIENT.send(httpRequest, BodyHandlers.ofString());
+        final String body = response.body();
+        final JiraIssue jiraIssue = GSON.fromJson(body, JiraIssue.class);
+        return jiraIssue;
+    }
+
+    private static URI getJiraIssueRestUri(final String issueId) throws URISyntaxException {
+        return new URI(JIRA_REST_BASE_URL + "issue/" + issueId);
+    }
+
+
+    public static boolean updateJiraFixVersion(String issueId, String gitHubMilestone)
+        throws URISyntaxException, IOException, InterruptedException {
+        final URI uri = getJiraIssueRestUri(issueId);
+        final UpdateJiraIssue updateJiraIssue = new UpdateJiraIssue(new UpdateFields(
+            jiraFixVersionFromGitHubMilestone(gitHubMilestone)));
+        final String json = new Gson().toJson(updateJiraIssue);
+        final HttpRequest httpRequest = authorizedRequestBuilder()
+            .uri(uri)
+            .PUT(BodyPublishers.ofString(json))
+            .header("Content-type", "application/json")
+            .build();
+        final HttpResponse<String> response = HTTP_CLIENT.send(httpRequest, BodyHandlers.ofString());
+        return response.statusCode() < HttpURLConnection.HTTP_MULT_CHOICE;
+    }
+
+    private static FixVersion[] jiraFixVersionFromGitHubMilestone(final String gitHubMilestone) {
+        if (gitHubMilestone == null) {
+            return new FixVersion[0];
+        }
+        return new FixVersion[] {new FixVersion(gitHubMilestoneToJiraVersion(gitHubMilestone))};
+    }
+
+    public static boolean updateGitHubMilestone(int number, String jiraFixVersion)  {
+        try {
+            final GHIssue issue = getDockstoreRepository().getIssue(number);
+            final String ghMilestoneDesc = jiraVersionToGitHubMilestone(jiraFixVersion);
+            final PagedIterable<GHMilestone> ghMilestones =
+                getDockstoreRepository().listMilestones(GHIssueState.ALL);
+            final Optional<GHMilestone> milestone = ghMilestones.toList().stream()
+                .filter(ghMilestone -> ghMilestoneDesc.equals(ghMilestone.getTitle()))
+                .findFirst();
+            if (milestone.isEmpty()) {
+                System.err.println("Could not find GitHub milestone for %s".formatted(jiraFixVersion));
+                return false;
+            }
+            issue.setMilestone(milestone.get());
+        } catch (IOException e) {
+            LOG.error("Error setting milestone on issue %s".formatted(number), e);
+            System.err.println("Error updating milestone for GitHub issue %s".formatted(number));
```

Comment:
Don't think you need this `System.err.println` since you have a `LOG.error` above

Reply:
I'd intentionally left it and other `System.err`s on the theory that they are user facing, and the log messages aren't always as clear for the actual error given the timestamp and class name being printed out. We do this (or did it?) in the CLI; you can turn off logging but still get error messages.

But on further reflection and looking at your comments, given that this is for internal use only anyway, I'll just use LOG across the board.